### PR TITLE
upcoming: [M3-8378] - OBJ Cleanup

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Related
- [x] https://github.com/linode/manager/pull/10797

## Changes  🔄
- Removed our react hook form logic for `BucketRateLimitTable` and corresponding files. This is far from being ready and it introduced complexity since this component is used in 3 different places. This will most likely be a 2025 initiative to add this logic in when it's ready.
    - By removing this, we fixed an issue where app was crashing when trying to display the bucket rate table in any of the drawers for E2/E3. 
- Moved the "Bucket Rate Limits" title and corresponding helper text into `BucketRateLimitTable.tsx` since we were repeating that in 3 different places.
- The Create Bucket drawer now displays each additional step once we have the preceding information selected.
   - Once we select a region, we'll show endpoint types
   - Once we select endpoint types, we'll show bucket rate limits (except for E0/E1 of course) 

## Target release date 🗓️
N/A

### Prerequisites
- Enable multi-cluster and gen2 flags
- Enable MSW

### Reproduction steps
(How to reproduce the issue, if applicable)
- When selecting a E2/E3 endpoint type bucket, app would crash because we didn't implement our form logic to `BucketDetailsDrawer`

### Verification steps
(How to verify changes)
- Pull down branch or preview link

- Select "Details" for any E2/E3 endpoint bucket:
   - Drawer should open and bucket rate limits table should show

- Create a bucket and observe:
   - Once a region is selected, endpoint types show
   - Once we select endpoint types, bucket rate limits show (except for E0/E1 of course) 

- Go to bucket details page and then properties tab
   - Ensure title, description, and table all look good
   - Do this for E0 -> E3 endpoint types

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by pmakode-akamai — APPROVED** [review]
Nice work with the clean-up 👍. Good catch on removing the react-hook-form logic for now.

- Removed react-hook-form logic as planned ✅
- No crash on details/create drawer page ✅
- Rate limit table visible only for E2 and E3 endpoint types ✅
- The Create Bucket drawer now displays each additional step once we have the preceding information selected ✅


Apart from the above, I observed a strange behavior: when the `OBJ Gen 2` flag is enabled, reloading the application while on the SSL tab or Properties tab doesn't load the data (with mocked or real data). 

I noticed:
```
const { data: bucketsData } = useObjectStorageBuckets(isObjectStorageGen2Enabled);
```
in `BucketDetail/index.tsx` at [Line 53](https://github.com/linode/manager/blob/579d4181013a189080b3f23a326101aff631d9bb/packages/manager/src/features/ObjectStorage/BucketDetail/index.tsx#L53) returns `undefined` when isObjectStorageGen2Enabled is explicitly set after reloading the application on those tabs . The function defaults to true if the flag is not passed, and it works fine in that case. I’m not sure if others are observing the same behavior or if it’s due to account capabilities for me.

Screenshot 2024-08-30 at 4 21 54 PM
----
Screenshot 2024-08-30 at 4 21 01 PM

And the same in the case of Properties tab, too.

**Comment by jaalah-akamai** [comment]
> Apart from the above, I observed a strange behavior: when the OBJ Gen 2 flag is enabled, reloading the application while on the SSL tab or Properties tab doesn't load the data (with mocked or real data).

Yea that's expected because there's no real data yet, it's all mocked. Also when you refresh on the tab, our mocked bucket labels change since they're randomized and the bucket you were looking at no longer exists. See [serverHandlers.ts#L979-L985](https://github.com/linode/manager/blob/develop/packages/manager/src/mocks/serverHandlers.ts#L979-L985)

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx [hunk 1]
```diff
@@ -50,66 +55,93 @@ const tableData = ({ endpointType }: BucketRateLimitTableProps) => {
...
+            <SupportLink
+              text="open a support ticket"
+              title="Request to Increase Bucket Rate Limits"
+            />
+            .{' '}
+          </>
+        ) : (
+          'This endpoint type supports up to 750 Requests Per Second (RPS). '
+        )}
+        Understand <Link to="#">bucket rate limits</Link>.
```
Comment (mjac0bs) [inline packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx:87]: Two questions for my own knowledge, not directly related to these PR's changes:

1. Have we moved away from using "Learn more about [topic]." in our copy? This is the first time I remember seeing "Understand [topic]." and it's used multiple times. I was wondering whether that's intentional or due to some inconsistency in tech writing.

Screenshot 2024-08-30 at 7 44 50 AM

2. Why is listed twice in this dropdown?
Screenshot 2024-08-30 at 7 44 59 AM

**Review by mjac0bs — COMMENTED** [review]
Confirmed:

✅  When selecting "Details" for any E2/E3 endpoint bucket:
   - Drawer should open and bucket rate limits table should show

✅ When creating a bucket:
   - Once a region is selected, endpoint types show
   - Once we select endpoint types, bucket rate limits show (except for E0/E1 of course)

❓ On the bucket details page Properties tab for E0-E3 endpoint types:
   - Ensure title, description, and table all look good

This is what I saw on the Properties tab of E0 and E1 buckets. There seems to be a `Save` button, although there's nothing to Save. Is that unexpected? (The drawer looks fine.)
| Tab | Drawer |
| --- | --- |
| Screenshot 2024-08-30 at 7 48 09 AM | Screenshot 2024-08-30 at 7 48 00 AM |

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx:87]: 1. I'll confirm with TW
2. In rare cases, the dropdown must display a specific endpoint hostname (s3_endpoint) along with the endpoint type to distinguish between two assigned endpoints of the same type. This is necessary for multiple Gen1 (E1) assignments in the same region. Essentially, this mock demonstrates that if an s3_endpoint is provided, we want to display it.

File: packages/manager/src/features/ObjectStorage/BucketDetail/BucketProperties.tsx [hunk 2]
```diff
@@ -23,69 +17,36 @@ interface Props {
...
-  const onSubmit = () => {
-    // TODO: OBJGen2 - Handle Bucket Rate Limit update logic once the endpoint for updating is available.
-    // The 'data' argument is expected -> data: UpdateBucketRateLimitPayload
-  };
-
   return (
-    <FormProvider {...form}>
+    <>
       <BucketBreadcrumb bucketName={label} history={history} prefix={prefix} />
       <StyledText>{hostname || 'Loading...'}</StyledText>
```
Comment (abailly-akamai) [inline packages/manager/src/features/ObjectStorage/BucketDetail/BucketProperties.tsx:31]: Something to fix for later IMO. An undefined hostname does necessarily means data is loading. Error handling and proper loading handlers should be considered when this is ready

**Review by abailly-akamai — APPROVED** [review]
I am lacking some context about the feature but PR is doing what it describes and fix the issue mentioned ✅

File: packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx:87]: TW wants to keep it as is even though we use it in one other place

Comment (mjac0bs) [reply packages/manager/src/features/ObjectStorage/BucketLanding/BucketRateLimitTable.tsx:87]: Thanks for confirming and for the dropdown explanation. Any verdict on the `Save` button?

**Pull request merged**